apps/api/src/app/workflows-v2/usecases/build-variable-schema/build-available-variable-schema.usecase.ts

```diff
@@ -38,17 +39,17 @@ export class BuildAvailableVariableSchemaUsecase {
           additionalProperties: false,
         },
         steps: buildPreviousStepsSchema(previousSteps, workflow.payloadSchema),
-        payload: safePayloadSchema(workflow) || { type: 'object', description: 'Payload for the current step' },
+        payload: safePayloadSchema(workflow),
       },
       additionalProperties: false,
     } as const satisfies JSONSchemaDto;
   }
 }
-function safePayloadSchema(workflow: NotificationTemplateEntity): JSONSchemaDto | undefined {
+function safePayloadSchema(workflow: NotificationTemplateEntity): JSONSchemaDto {
   try {
-    return JSON.parse(workflow.payloadSchema);
+    return parsePayloadSchema(workflow.payloadSchema);
   } catch (e) {
-    return undefined;
+    return { type: 'object', description: 'Payload for the current step' };
```

Comment:
Do we need this hardcoded schema object or should we just return `{}` or `null` if parse fails? The latter feels safer and more future proof.

Reply:
I agree it should merge the logic, but the default is important, as it's the schema eq to {}, you are stating a general object with no additional information in the schema, having no payload schema doesn't mean you acannot use it as a variable